plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/PdeMain.java

```diff
@@ -0,0 +1,381 @@
+            .serviceCache(ArServiceRecordCache.withEntryLifetimeLimit(Duration.ZERO))
+            .localHostnamePort(pdeHostname, pdePort)
+            .plugins(new HttpJsonCloudPlugin.Builder()
+                .orchestrationStrategy(strategy)
+                .serviceRegistrySocketAddress(serviceRegistryAddress)
+                .build());
+
+        final boolean secureMode = Boolean.parseBoolean(getProp(appProps, "server.ssl.enabled"));
+
+        if (secureMode) {
+            final String trustStorePath = getProp(appProps, "server.ssl.pde.trust-store");
+            final char[] trustStorePassword = getProp(appProps, "server.ssl.pde.trust-store-password").toCharArray();
+            final String keyStorePath = getProp(appProps, "server.ssl.pde.key-store");
+            final char[] keyPassword = getProp(appProps, "server.ssl.pde.key-password").toCharArray();
+            final char[] keyStorePassword = getProp(appProps, "server.ssl.pde.key-store-password").toCharArray();
+            systemBuilder.identity(loadIdentity(keyStorePath, keyPassword, keyStorePassword))
+                .trustStore(loadTrustStore(trustStorePath, trustStorePassword));
+        } else {
+            systemBuilder.name(PDE_SYSTEM_NAME).insecure();
+        }
+
+        return systemBuilder.build();
+
+    }
+
+    /**
+     * Loads the keystore at the given path, using the given password.
+     * <p>
+     * {@code path} is first treated as a regular file path. If the certificate
+     * cannot be found at that location, an attempt is made to load it from
+     * resources (i.e. within the jar file).
+     * If the keystore cannot be loaded, the application is terminated.
+     *
+     * @param path     Keystore path.
+     * @param password Password or private key associated with the
+     *                 keystore.
+     * @return The loaded keystore.
+     */
+    private static KeyStore loadKeyStore(String path, char[] password) {
+
+        KeyStore keyStore = null;
+        try {
+            keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        } catch (KeyStoreException e) {
+            logger.error("Failed to load identity keystore", e);
+            System.exit(74);
+        }
+
+        File keyStoreFile = new File(path);
+
+        if (keyStoreFile.isFile()) {
+            try (FileInputStream in = new FileInputStream(keyStoreFile)) {
+                keyStore.load(in, password);
+            } catch (NoSuchAlgorithmException | CertificateException | IOException e) {
+                logger.error("Failed to read keystore from file", e);
+                System.exit(74);
+            }
+        } else {
+            try (InputStream in = ClassLoader.getSystemResourceAsStream(path)) {
+                keyStore.load(in, password);
+            } catch (NoSuchAlgorithmException | CertificateException | IOException e) {
+                logger.error("Failed to load keystore from resources", e);
```

Comment:
typo: resources => resource

Reply:
Clarified (changed to _resources directory_